### PR TITLE
Added workflows for github release

### PR DESCRIPTION
**Pull request by sajayantony**
Signed-off-by: Sajay Antony <sajaya@microsoft.com>

Changes are based off github.com/deislabs/ratify 

Sample release can be found here 
https://github.com/sajayantony/notation-azure-kv/releases/tag/untagged-6851f22ba197c4a88dc6

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: Release notation-azure-kv
+
+on:
+  push:
+    tags:
+      - 'v*'
+
```
Comment (sozercan) [inline .github/workflows/release.yml:7]: make sure to define required permissions https://github.blog/changelog/2021-04-20-github-actions-control-permissions-for-github_token/

**Review by sozercan — APPROVED** [review]

File: .goreleaser.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      - linux
+      - windows
+    goarch:
+      - amd64
+      - arm64
+    ignore:
+      - goos: windows
+        goarch: arm64
+    ldflags:
+      - -w -X github.com/azure/notation-azure-kv/internal/version.GitTag={{.Version}} -X github.com/azure/azure-notation-kv/internal/version.GitCommitHash={{.FullCommit}}
```
[the remark this answers is not in the document]
Comment (sajayantony) [reply .goreleaser.yml]: Please verify if the flags look good.

File: .github/workflows/release.yml (hunk 1 above)
Comment (sajayantony) [reply .github/workflows/release.yml:7]: Added permissions section below for release. We should add a PR builder workflow as well but this won't use any tokens.

Comment (aramase) [reply .github/workflows/release.yml:7]: We should change the permission here to `Read repository contents permission` too as the default is read and write.
Screen Shot 2021-12-14 at 11 15 29 AM

File: .goreleaser.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      - linux
+      - windows
+    goarch:
+      - amd64
+      - arm64
+    ignore:
+      - goos: windows
+        goarch: arm64
+    ldflags:
+      - -s -w -X {{.ModulePath}}/internal/version.Version={{.Version}} -X {{.ModulePath}}/internal/version.BuildMetadata=
```
Comment (aramase) [inline .goreleaser.yml:22]: Where is the `ModulePath` defined?

File: .github/workflows/release.yml (hunk 1 above)
Comment (sajayantony) [reply .github/workflows/release.yml:7]: Given that this is creating a release, its needs release permissions is what I observed. If there is another workflow in another repo that works for creating release, could you point me to it and I can make the updates.

File: .goreleaser.yml (hunk 3 above)
Comment (aramase) [reply .goreleaser.yml]: xref: https://goreleaser.com/customization/templates/

.ModulePath  - the go module path, as reported by `go list -m`

Comment (sajayantony) [reply .goreleaser.yml]: This is from the goreleaser template https://goreleaser.com/customization/templates/

**Review by aramase — APPROVED** [review]

**Comment by sajayantony** [comment]
Fixed missing commit id in ff2c8bf212499b16c2ef8fe47c74cabcfb949814

**Pull request merged**